x/globalfee/types/params.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// ParamStoreKeyMinGasPrices store key
+var ParamStoreKeyMinGasPrices = []byte("MinimumGasPricesParam")
+
+// DefaultParams returns default parameters
+func DefaultParams() Params {
+	return Params{MinimumGasPrices: sdk.DecCoins{}}
+}
+
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// ValidateBasic performs basic validation.
+func (p Params) ValidateBasic() error {
+	return validateMinimumGasPrices(p.MinimumGasPrices)
+}
+
+// ParamSetPairs returns the parameter set pairs.
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(
+			ParamStoreKeyMinGasPrices, &p.MinimumGasPrices, validateMinimumGasPrices,
+		),
+	}
+}
+
+// this requires the fee non-negative
+func validateMinimumGasPrices(i interface{}) error {
+	v, ok := i.(sdk.DecCoins)
+	if !ok {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "type: %T, expected sdk.DecCoins", i)
+	}
+
+	dec := DecCoins(v)
+	return dec.Validate()
+}
+
+// Validate checks that the DecCoins are sorted, have nonnegtive amount, with a valid and unique
+// denomination (i.e no duplicates). Otherwise, it returns an error.
+type DecCoins sdk.DecCoins
```

Comment:
Why need to re-write this https://github.com/cosmos/cosmos-sdk/blob/v0.46.12/types/dec_coin.go#L512?

Reply:
interesting. Gaia also re-write this. 
let me check some edge cases on isPositive and isNegative.

Reply:
think it's ok. removed and use the original one.